server/test/new_objects.go

```diff
@@ -106,6 +106,74 @@ func AddAllHostsLabel(t *testing.T, ds fleet.Datastore) {
 	require.NoError(t, err)
 }
 
+func AddBuiltinLabels(t *testing.T, ds fleet.Datastore) {
```

Comment:
I understand the idea of this, but it will be easy to get out-of-sync with the DB migrations (add or modify a built-in label and forget to update this). I _think_ it might be ok though given that `mysql.CreateNamedMySQLDS` runs the `table` migrations (via the schema.sql), and we shouldn't use the `data` migrations anymore, so new builtin-labels should be automatically created in the future.

Reply:
> so new builtin-labels should be automatically created in the future.

@mna, I'm not sure I understand what you're saying here. The issue that prompted this addition was that it seemed at least in some tests only the "All Hosts" label existed and the datastore did not include any of the other builtins. If  `table` migrations include `data` operations (inserting a new row to the existing `labels`), are those going to be reflected in the `schema.sql`? From what I observed, I don't think that is the case, but I might be missing something. 

> add or modify a built-in label and forget to update this

This was definitely on my mind too, which is why I included a check against `fleet.ReservedLabelNames` in this method to try to catch cases where the `fleet` package and the developer forgot to update this too. Your comment also makes me think it might be good to also include an ad hoc query of all builtins `SELECT * FROM LABELS WHERE label_type = 1` and check that the names and queries match the expected set.

Reply:
@gillespi314 Sorry, I wasn't super clear and the names are misleading - we have 2 types of DB migrations, `data` and `tables`. The `data` ones used to be used to insert "permanent" data in the database, and `tables` for create/alter tables (DML vs DDL). However, [according to the `README`](https://github.com/fleetdm/fleet/tree/main/server/datastore/mysql/migrations/data#data-migrations) we shouldn't use the `data` migrations anymore, even when we want to insert rows we should use a `tables` migration.

I believe that `schema.sql` only contains the results of `tables` migrations, so while it doesn't have the old builtin labels (as they were `data` migrations), new ones (including the new one for OS Updates) will be `tables` migrations and should be part of the schema.

> This was definitely on my mind too, which is why I included a check against fleet.ReservedLabelNames in this method 

I missed that! Great idea.

> it might be good to also include an ad hoc query of all builtins SELECT * FROM LABELS WHERE label_type = 1 and check that the names and queries match the expected set.

Yeah as you can see from the `data` vs `tables` migrations, you'd only validate that the new (`tables` migrations) builtins are there, the old ones wouldn't be in the DB before the ones added by this function. I think with the `ReservedLabelNames` it's a good-enough check for now and we can revisit if we find it too error-prone in the future? I'll make a mental note to add the new label name in that list!

Reply:
Thanks! And it looks like `schema.sql` does include DML operations (e.g., `INSERT`) so that was a mistaken assumption on my part :)

> we shouldn't use the data migrations anymore, even when we want to insert rows we should use a tables migration.

As an aside, we may want to think about ways to better communicate/enforce the "should". The `chrome` label was added after the `README` (which is why I assumed the schema.sql wasn't reflecting the DML operations) but it was added as a `data` [migration](https://github.com/fleetdm/fleet/blob/bff50c5ced883e5aa28d48290b64dedf3212ed26/server/datastore/mysql/migrations/data/20230525175650_InsertChromeBuiltInLabel.go#L21-L22) 🙃

Reply:
This could probably be a CI check (e.g. fail if there are git changes in this directory, with a message that points to the README).